nnmnkwii/datasets/jvs.py

```diff
@@ -0,0 +1,221 @@
+# coding: utf-8
+
+from __future__ import with_statement, print_function, absolute_import
+
+from nnmnkwii.datasets import FileDataSource
+
+import numpy as np
+import os
+from os import listdir
+from os.path import join, splitext, isdir, exists, basename, isfile
+from glob import glob
+from collections import OrderedDict
+
+available_speakers = ["jvs" + str(i).zfill(3) for i in range(1, 101)]
+# Three lost wavfiles in this dataset
+lost_wavfiles = {
+    "jvs089": ["VOICEACTRESS100_019"],
+    "jvs030": ["VOICEACTRESS100_045"],
+    "jvs074": ["VOICEACTRESS100_094"]}
+
+
+def _parse_speaker_info(data_root):
+    # only use gender_f0range.txt
+    speaker_info_path = join(data_root, "gender_f0range.txt")
+    if not exists(speaker_info_path):
+        raise RuntimeError("File {} doesn't exist".format(speaker_info_path))
+    speaker_info = OrderedDict()
+    terms = ["speaker", "Male_or_Female", "minf0[Hz]", "maxf0[Hz]"]
+    with open(speaker_info_path, "r", encoding="utf8") as file_:
+        for line in file_:
+            fields = line.strip().split()
+            if fields[0] == terms[0]:
+                continue
+            assert len(fields) == 4
+            speaker, gender, minf0, maxf0 = fields
+            speaker_info[speaker] = {}
+            speaker_info[speaker]["gender"] = gender
+            speaker_info[speaker]["minf0"] = minf0
+            speaker_info[speaker]["maxf0"] = maxf0
+    return speaker_info
+
+
+class _JVSBaseDataSource(FileDataSource):
+    def __init__(self, data_root, speakers, labelmap, max_files):
+        # only accept "jvs*" format
+        self.data_root = data_root
+        if speakers == "all":
+            speakers = available_speakers
+        for speaker in speakers:
+            if speaker not in available_speakers:
+                raise ValueError("Unknown speaker {}. It should be one of {}".format(speaker, available_speakers))
+        self.speakers = speakers
+        if labelmap is None:
+            labelmap = {s: idx for idx, s in enumerate(speakers)}
+        self.labelmap = labelmap
+        self.labels = None
+        self.max_files = max_files
+        # TODO: currently there are many problem in falset, so we do not support falset
+        self._folders = ["parallel100", "nonpara30", "whisper10"]
+        self._textfilename = "transcripts_utf8.txt"
+        self._wavfoldername = "wav24kHz16bit"
+        self.speaker_info = _parse_speaker_info(data_root)
+        self._nonpara_no_wav = self._validate()
```

Comment:
It's a bit hard to understand what nonpara_no_wav is. Would be better to have a more descriptive name + docstrings.

Reply:
I change to nonpara_without_name,
since some of the text in nonpara data have no corresponding wav file, so we need to record it